recipes/openssl/1.x.x/conanfile.py

```diff
@@ -700,16 +715,34 @@ def package_info(self):
         self.cpp_info.names["cmake_find_package"] = "OpenSSL"
         self.cpp_info.names["cmake_find_package_multi"] = "OpenSSL"
         if self._use_nmake:
+            libsuffix = "d" if self.settings.build_type == "Debug" else ""
             if self._full_version < "1.1.0":
-                self.cpp_info.libs = ["ssleay32", "libeay32"]
+                self.cpp_info.components["ssl"].libs = ["ssleay32"]
+                self.cpp_info.components["crypto"].libs = ["libeay32"]
             else:
-                if self.settings.build_type == "Debug":
-                    self.cpp_info.libs = ['libssld', 'libcryptod']
-                else:
-                    self.cpp_info.libs = ['libssl', 'libcrypto']
+                self.cpp_info.components["ssl"].libs = ["libssl" + libsuffix]
+                self.cpp_info.components["crypto"].libs = ["libcrypto" + libsuffix]
         else:
-            self.cpp_info.libs = ["ssl", "crypto"]
+            self.cpp_info.components["ssl"].libs = ["ssl"]
+            self.cpp_info.components["crypto"].libs = ["crypto"]
+ 
+        self.cpp_info.components["ssl"].requires = ["crypto"]
+
+        if self._full_version < "1.1.0" and not self.options.get_safe("no_zlib"):
+            self.cpp_info.components["crypto"].requires = ["zlib::zlib"]
+
         if self.settings.os == "Windows":
-            self.cpp_info.system_libs.extend(["crypt32", "msi", "ws2_32", "advapi32", "user32", "gdi32"])
+            self.cpp_info.components["crypto"].system_libs.extend(["crypt32", "ws2_32", "advapi32", "user32"])
         elif self.settings.os == "Linux":
-            self.cpp_info.system_libs.extend(["dl", "pthread"])
+            self.cpp_info.components["crypto"].system_libs.append("dl")
+            self.cpp_info.components["ssl"].system_libs.append("dl")
+            if not self.options.no_threads:
+                self.cpp_info.components["crypto"].system_libs.append("pthread")
+                self.cpp_info.components["ssl"].system_libs.append("pthread")
+
+        self.cpp_info.components["crypto"].names["cmake_find_package"] = "Crypto"
+        self.cpp_info.components["crypto"].names["cmake_find_package_multi"] = "Crypto"
+        self.cpp_info.components["crypto"].names['pkg_config'] = 'libcrypto'
```

Comment:
Components are not yet available for `pkg_config` generator. I'm opening the issue here (https://github.com/conan-io/conan/issues/7330), what is the expected result from this declaration?

Reply:
Hum... I think that the idea was more something like this :
`self.cpp_info.names['pkg_config'] = 'libcrypto'`

Reply:
I would like to generate thoses files which are generated by openssl build.
```
./lib/pkgconfig/openssl.pc
./lib/pkgconfig/libssl.pc
./lib/pkgconfig/libcrypto.pc
```
If it is not ready not, I don't think it is an issue, is it ?

Reply:
Correct, see the `.pc` files at https://fedora.pkgs.org/30/fedora-x86_64/openssl-devel-1.1.1b-3.fc30.x86_64.rpm.html

Reply:
No, `pkg_config` doesn't understand components yet. The issue  is open here: https://github.com/conan-io/conan/issues/7330